### PR TITLE
Adds interactions for exoplanet foliage and terrain

### PR DESCRIPTION
**Pull request by kyres1**
- rscadd: "Added a method of clearing paths into exoplanet land by using mini hoes."
  - tweak: "Scythes and hatchets can now clear grass and bushes on exoplanets."
  - bugfix: "Changed the paths of jungle decals to be structures instead, so grass can actually be interacted with."

  Obligatory among us :
image

### CODE REVIEW

**Comment by kyres1** [comment]
!review

File: code/modules/overmap/exoplanets/decor/_turfs.dm [hunk 1]
```diff
@@ -43,6 +43,14 @@
 		if(T.use(1))
 			playsound(src, 'sound/items/Deconstruct.ogg', 80, 1)
 			ChangeTurf(/turf/simulated/floor, FALSE, FALSE, FALSE, TRUE)
+	else if(diggable && istype(C,/obj/item/material/minihoe))
+		visible_message("<span class='notice'>\The [user] starts clearing \the [src]</span>")
```
Comment (FluffyGhoster) [inline code/modules/overmap/exoplanets/decor/_turfs.dm:47]:
```suggestion
		visible_message(SPAN_NOTICE("\The [user] starts clearing \the [src]"))
```

File: code/modules/overmap/exoplanets/decor/_turfs.dm [hunk 2]
```diff
@@ -43,6 +43,14 @@
...
+		if(C.use_tool(src, user, 50, volume = 50))
+			to_chat(user,"<span class='notice'>You make a small clearing.</span>")
```
Comment (FluffyGhoster) [inline code/modules/overmap/exoplanets/decor/_turfs.dm:49]:
```suggestion
			to_chat(user, SPAN_NOTICE("You make a small clearing."))
```

File: code/modules/overmap/exoplanets/decor/_turfs.dm [hunk 3]
```diff
@@ -43,6 +43,14 @@
...
+			new /obj/structure/clearing(src)
+			diggable = 0
+		else
+			to_chat(user,"<span class='notice'>You stop shoveling.</span>")
```
Comment (FluffyGhoster) [inline code/modules/overmap/exoplanets/decor/_turfs.dm:53]:
```suggestion
			to_chat(user, SPAN_NOTICE("You stop shoveling."))
```

File: code/modules/overmap/exoplanets/decor/flora/_bush.dm [hunk 4]
```diff
@@ -1,3 +1,34 @@
...
+			if(prob(40))
+				new /obj/item/stack/material/wood(get_turf(src), 4)
+			if(prob(10))
+				var/pickberry = pick(list(/obj/item/seeds/berryseed, /obj/item/seeds/blueberryseed))
+				new /obj/item/stack/material/wood(get_turf(src), 4)
+				new pickberry(get_turf(src), 4)
+				to_chat(user, SPAN_NOTICE("You find some seeds as you hack the bush away."))
+			to_chat(user, SPAN_NOTICE("You slice at the bush!"))
+			qdel(src)
+	if(istype(W, /obj/item/material/hatchet))
```
Comment (FluffyGhoster) [inline code/modules/overmap/exoplanets/decor/flora/_bush.dm:27]:
```suggestion
	else if(istype(W, /obj/item/material/hatchet))
```

File: code/modules/overmap/exoplanets/decor/flora/snow.dm [hunk 5]
```diff
@@ -5,6 +5,18 @@
 	icon = 'icons/obj/flora/snowflora.dmi'
 	density = FALSE
 
+/obj/structure/flora/grass/attackby(var/obj/item/W, var/mob/user)
+	if(istype(W, /obj/item/material/scythe))
+		shake_animation()
+		if(W.use_tool(src, user, 10, volume = 50))
+			to_chat(user, SPAN_NOTICE("You slice the grass away with the scythe!"))
+			qdel(src)
+	if(istype(W, /obj/item/material/hatchet))
```
Comment (FluffyGhoster) [inline code/modules/overmap/exoplanets/decor/flora/snow.dm:14]:
```suggestion
	else if(istype(W, /obj/item/material/hatchet))
```

File: code/modules/overmap/exoplanets/decor/objs/clearing.dm [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+		/turf/simulated/floor/tiled,
+		/turf/simulated/floor/wood,
+		/turf/simulated/wall
+	)
+
+/obj/structure/clearing/attackby(obj/item/W, mob/user)
+	if(istype(W,/obj/item/material/minihoe))
+		visible_message("<span class='notice'>\The [user] starts adjusting the \the [src]</span>")
+		if(W.use_tool(src, user, 50, volume = 50))
+			visible_message("<span class='notice'>\The [user] adjusts \the [src]!</span>")
```
Comment (FluffyGhoster) [inline code/modules/overmap/exoplanets/decor/objs/clearing.dm:25]:
```suggestion
			visible_message(SPAN_NOTICE("\The [user] adjusts \the [src]!"))
```

File: code/modules/overmap/exoplanets/decor/objs/clearing.dm [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+		/turf/simulated/floor/concrete,
+		/turf/simulated/floor/plating,
+		/turf/simulated/floor/tiled,
+		/turf/simulated/floor/wood,
+		/turf/simulated/wall
+	)
+
+/obj/structure/clearing/attackby(obj/item/W, mob/user)
+	if(istype(W,/obj/item/material/minihoe))
+		visible_message("<span class='notice'>\The [user] starts adjusting the \the [src]</span>")
```
Comment (FluffyGhoster) [inline code/modules/overmap/exoplanets/decor/objs/clearing.dm:23]:
```suggestion
		visible_message(SPAN_NOTICE("\The [user] starts adjusting the \the [src]"))
```

File: code/modules/overmap/exoplanets/decor/objs/clearing.dm [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+		if(W.use_tool(src, user, 50, volume = 50))
+			visible_message("<span class='notice'>\The [user] adjusts \the [src]!</span>")
+			qdel(src)
+		else
+			to_chat(user, "<span class='notice'>You stop adjusting the terrain.</span>")
```
Comment (FluffyGhoster) [inline code/modules/overmap/exoplanets/decor/objs/clearing.dm:28]:
```suggestion
			to_chat(user, SPAN_NOTICE("You stop adjusting the terrain."))
```

File: code/modules/overmap/exoplanets/decor/_turfs.dm [hunk 9]
```diff
@@ -43,6 +43,14 @@
 		if(T.use(1))
 			playsound(src, 'sound/items/Deconstruct.ogg', 80, 1)
 			ChangeTurf(/turf/simulated/floor, FALSE, FALSE, FALSE, TRUE)
+	else if(diggable && istype(C,/obj/item/material/minihoe))
+		visible_message("<span class='notice'>\The [user] starts clearing \the [src]</span>")
+		if(C.use_tool(src, user, 50, volume = 50))
+			to_chat(user,"<span class='notice'>You make a small clearing.</span>")
+			new /obj/structure/clearing(src)
+			diggable = 0
```
Comment (SleepyGemmy) [inline code/modules/overmap/exoplanets/decor/_turfs.dm:51]:
```suggestion
			diggable = FALSE
```

**Review by NonQueueingMatt — APPROVED** [review]

**Review by FluffyGhoster — APPROVED** [review]

**Pull request merged**